src/components/dashboard/main-content/course-enrollments/course-cards/BaseCourseCard.jsx

```diff
@@ -341,19 +334,14 @@ const BaseCourseCard = ({
     );
   };
 
-  const renderEmailSettingsModal = () => {
-    if (!hasEmailsEnabled) {
```

Comment:
The issue with removing `hasEmailsEnabled` stuff altogether is that not all course enrollment card types/variants should include the email settings dropdown. For example, assignment cards (`AssignedCourseCard`) should not expose the "Emails settings" dropdown given the assignment is not yet a legit course enrollment (i.e., learner hasn't accepted their assignment yet), so the user would likely face an error when trying to submit the email settings modal to save their choice.

As is, removing this logic, incorrectly exposes the "Email settings" dropdown item for assignments:

<img width="1213" alt="image" src="https://github.com/user-attachments/assets/4794668a-b159-4b45-a878-3c76aeb5d514">

When keeping this `hasEmailsEnabled` logic, it might be falsey for the following reasons:
1. `BaseCourseCard` is of type/variant `AssignedCourseCard`, where `hasEmailsEnabled` is `undefined`.
1. `BaseCourseCard` is _not_ of type/variant `AssignedCourseCard`, where `hasEmailsEnabled` represents the state of the whether the learners's current choice of whether emails are enabled for the course enrollment (i.e., `false`, `true`).

The modal previously wasn't opening when a learner has emails disabled for an enrollment because of the (2) case above, whereby `false` is falsey, fulfilling this condition and thus returning `null`.

We would only want to short-circuit the rendering of the `EmailSettingsModal` when `hasEmailsEnabled` is a legit, defined value (i.e., not `undefined`/`null`).

For all `BaseCourseCard` types/variants, the data is run through the `transformCourseEnrollment` function, and sets the `hasEmailsEnabled` property based on the `emailsEnabled` property for the given course enrollment. If the `emailsEnabled` property is `undefined`/`null` (i.e., when the "course enrollment" here is actually representing an assignment, where email settings is not applicable).

By removing the `hasEmailsEnabled` stuff altogether, notably within `getDropdownMenuItems`, the "Email settings" dropdown item is incorrectly exposed for `AssignedCourseCard`, which would introduce a different bug since the related email settings API would error when called within the modal.

I might suggest keeping all the `hasEmailsEnabled` stuff but make sure the `BaseCourseCard` component properly handles `hasEmailSettings` as `undefined`/`null` and `false` as separate falsey conditions throguhout to mitigate these issues. Consider using the `isDefinedAndNotNull` helper function ([source](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/42c06b0e09490c195e870d2c71cb7136bc976f9a/src/utils/common.js#L30-L33)), e.g.:

```jsx
const renderEmailSettingsModal = () => {
    if (!isDefinedAndNotNull(hasEmailsEnabled)) {
      return null;
    }
};
```